source/code/plugins/in_agent_telemetry.rb

```diff
@@ -0,0 +1,117 @@
+require 'fluent/input'
+require 'fluent/config/error'
+
+module Fluent
+
+  class Agent_Telemetry < Input
+    Plugin.register_input('agent_telemetry', self)
+
+    def initialize
+      super
+      require_relative 'agent_telemetry_script'
+    end
+
+    config_param :query_interval, :time, :default => '5m'
+    config_param :poll_interval, :time, :default => '15s'
+    config_param :omsadmin_conf_path, :string
+    config_param :cert_path, :string
+    config_param :key_path, :string
+    config_param :pid_path, :string
+    config_param :proxy_path, :string, :default => '/etc/opt/microsoft/omsagent/proxy.conf' #optional
+    config_param :os_info, :string, :default => '/etc/opt/microsoft/scx/conf/scx-release' #optional
+    config_param :install_info, :string, :default => '/etc/opt/microsoft/omsagent/sysconf/installinfo.txt' #optional
+
+    MIN_QUERY_INTERVAL = 1
+    MAX_QUERY_INTERVAL = 60 * 60 * 1
+
+    def configure (conf)
+      super
+      if !@omsadmin_conf_path
+        raise Fluent::ConfigError, "'omsadmin_conf_path' option is required on agent_telemetry input"
+      end
+      if !@cert_path
+        raise Fluent::ConfigError, "'cert_path' option is required on agent_telemetry input"
+      end
+      if !@key_path
+        raise Fluent::ConfigError, "'key_path' option is required on agent_telemetry input"
+      end
+      if !@pid_path
+        raise Fluent::ConfigError, "'pid_path' option is required on agent_telemetry input"
+      end
+    end
+
+    def start
+      @telemetry_script = OMS::Telemetry.new(@omsadmin_conf_path, @cert_path, @key_path,
+                                             @pid_path, @proxy_path, @os_info, @install_info, @log, false)
+
+      if @query_interval and @poll_interval
+        @query_finished = false
+        @poll_finished = false
+        @query_condition = ConditionVariable.new
+        @poll_condition = ConditionVariable.new
```

Comment:
@shkhisti I am now aware of the new issues with the heartbeats not sending on 1m interval (possibly due to ConditionVariable). Originally I just emulated the in_heartbeat_request pattern, I will alter this e.g. to use sleep instead.